cmake/Modules/GnuradioConfig.cmake.in

```diff
@@ -36,6 +36,8 @@ if(${ENABLE_PYTHON})
   set(PYTHON_EXECUTABLE @PYTHON_EXECUTABLE@)
   include(GrPython)
 endif()
+# pybind11 needed even without Python enabled for building against gnuradio-runtime
+find_dependency(pybind11)
```

Comment:
Hm, so it is not possible to build GNU Radio runtime withouth pybind11/Python? Not that I mind, but it's a bit surprising for the C++ interface to still require pybind11.

Reply:
I can't say I understand how this all works, but I think that it's a required dependency for `gnuradio-runtime` and thus all OOT modules because of the header inclusion in `block_gateway.h` [here](https://github.com/gnuradio/gnuradio/blob/1a0be2e6b54496a8136a64d86e372ab219c6559b/gnuradio-runtime/include/gnuradio/block_gateway.h#L18). I believe that is also the reason that it is necessary to build GNU Radio and any OOTs with the **same** version (or a compatible version) of pybind11, as has recently come up a few times on the mailing list. (Maybe there is something we should be doing better there with this CMake Config file to prevent bad builds). I haven't actually looked through the header or the code to see what exactly it needs pybind11 for.

Reply:
I like these changes to better export the targets, but we need to figure out why/if pybind11 is/has always been required for non-python builds and fix that too.

Reply:
The issue looks to be only `block_gateway`, which currently needs `pybind11` but it's also built into `gnuradio-runtime` regardless of if Python support is enabled. The commit where `pybind11` was introduced is 0249f7ce0cf5173b946b936c5cd66380afc8bf92, so thankfully we have someone here who should know what's going on there 😁. I haven't looked at this code to try to understand it, but I imagine `block_gateway` *could* be put behind `ENABLE_PYTHON`? Is there any way it could work or be rewritten for someone to build the runtime without `pybind11` but then still give modules (including runtime itself, I suppose) the option of building Python bindings?

Reply:
That is a quandary - similar setup we had had with block_gateway previously being dependent on swig director functionality - I'll have to go back and look at 3.8 to see how we were able to build gr without python/swig.  Maybe it just works now with pybind11 and no python - i just haven't tried it yet. Since pybind11 is just headers

Reply:
I suspect that it would work fine as-is now with just `pybind11` and no `ENABLE_PYTHON`. The caveat with needing `pybind11` in this manner is that it *does* introduce a version constraint on the `pybind11` that is later used to build the bindings. We've seen reports of this. AFAIK it doesn't have to be the exact same version, but only ones with the same `PYBIND11_INTERNALS_VERSION`. So it might be nice on a couple fronts to eliminate `block_gateway`'s dependence on `pybind11`, if it's possible.